get_block_details.py

```diff
@@ -0,0 +1,189 @@
+from lxml import html
+import requests
+import json
+from block import Block
+import logging
+
+# defines a function and stores value as 'title'
+def shorten_title(title):
+    """
+    Gets a PV title by shortening its address to the last segment.
+
+    Args:
+        title: The PV address as string.
+
+    Returns: The last segment of the input PV address as string.
+
+    """
+    # recieve the new value as a string for the RC value
+    # split string of the value at each colon
+    # look at each seperate string
+    # if there is a match on the block name
+    # new condition, if there is a match 'RC'
+    # new condition, if there is a value
+    # store it as the title
+    # else, move to the next block of code
+
+
+    # new variable declared as 'number'
+    # finding the last ':' in the 'title'
+    number = title.rfind(':')
+    # returns the title where ':' is found
+    return title[number + 1:]
+
+def shorten_rc_title(title):
+    """
+    Gets a PV title by shortening its address to the last segment.
+
+    Args:
+        title: The PV address as string.
+
+    Returns: The last segment of the input PV address as string.
+
+    """
+    # recieve the new value as a string for the RC value
+    # split string of the value at each colon
+    # look at each seperate string
+    # if there is a match on the block name
+    # new condition, if there is a match 'RC'
+    # new condition, if there is a value
+    # store it as the title
+    # else, move to the next block of code
+
+
+    # new variable declared as 'number'
+    # finding the last ':' in the 'title'
+
+    title = title.split(':')
+    rc_name = "RC"
+    value_name = ["HIGH.VAL", "LOW.VAL", "INRANGE.VAL", "ENABLED.VAL"]
+    title_list = []
+
+    if rc_name in title:
+        title_list.append(title[-3])
+        title_list.append(rc_name)
+        for name in value_name:
+            if name in title:
+                title_list.append(name)
+                title = ':'.join(title_list)
+                return title
+    else:
+        title = title[-1]
+        return title
+
+
+def get_rc_values_for_block(block_name, pvs):
+    """Search pvs for RC values for given block and return them"""
+    for k, v in pvs.items():
+        block_names = k.split(':')
+        if block_name == block_names[0]:
+            print block_name, v
+    return v
```

Comment:
This always returns the last value (`v`) in the dict of `pvs`. Worse still, this assumes dictionaries are ordered which is *sort of* true but not guaranteed in 2.7 and it completely not true in 3+. I assume we actually want to break as soon as we find a `block_name` that matches.